src/utils.py

```diff
@@ -336,16 +342,26 @@ def submission_formatting(submission):
     return prediction_formatted
 
 
-def calculate_map(metrics_filepath):
+def calculate_map(metrics_filepath, list_of_desired_classes=None, mappings_file=None):
     with open(metrics_filepath) as f:
         metrics = f.read().splitlines()
 
+    if list_of_desired_classes:
+        codes2names, names2codes = get_class_mappings(mappings_file)
+        if not all([cls.startswith('/') for cls in list_of_desired_classes]):
+            list_of_desired_classes = [names2codes.get(cls_name, 'notfound')
+                                       for cls_name in list_of_desired_classes]
```

Comment:
If you create list_of_desired_classed but you use it without checking that you have only valid labels (like the second assert in reduce_number_of_classes) you risk passing a list with "nolabel" inside (not sure if thats critical)

Reply:
@i008 added assertion